app/controllers/blackouts_controller.rb

```diff
@@ -93,10 +93,30 @@ def create # rubocop:disable MethodLength
   end
 
   def update
+    p = blackout_params
+    p[:created_by] = current_user.id 
     @blackout.set_id = nil
-    if @blackout.update_attributes(blackout_params)
-      redirect_to @blackout, notice: 'Blackout was successfully updated.'
+
+    # check for conflicts
+    res = Reservation.overlaps_with_date_range(p[:start_date], p[:end_date])
+          .active
+
+    # save and exit
+    if res.empty? && @blackout.update_attributes(p)
```

Comment:
I'm a little hesitant to do both of these at the same time -- I think we should only attempt to update the blackout if there aren't any conflicting reservations.

This is also a great opportunity to create a [service object](http://multithreaded.stitchfix.com/blog/2015/06/02/anatomy-of-service-objects-in-rails/) to handle blackout updating. PR #1624 has an example of a service object for reservation creation. 

Essentially, this method should only get the params and pass them to the service object, which will handle the update process and return the appropriate messages (e.g. errors or the updated object)